src/main/java/org/jahia/modules/sitemap/filter/SitemapCacheFilter.java

```diff
@@ -59,126 +54,85 @@
 
     private static final Logger logger = LoggerFactory.getLogger(SitemapCacheFilter.class);
 
+    private static final String PREFIX_CACHE_NAME = "sitemap-cache-";
+    private Ehcache sitemapCache;
+    private SitemapService sitemapService;
+
+    @Reference
+    public void setSitemapService(SitemapService sitemapService) {
+        sitemapCache = sitemapService.getSitemapEhCache();
```

Comment:
you need to choose between moving all the logic into the service (better option) then add and use service methods to manipulate the cache, or set up the cache here